src/UnderlineNav2/UnderlineNav.tsx

```diff
@@ -120,15 +116,15 @@ const getValidChildren = (children: React.ReactNode) => {
 
 const calculatePossibleItems = (childWidthArray: ChildWidthArray, navWidth: number, moreMenuWidth = 0) => {
   const widthToFit = navWidth - moreMenuWidth
-  let breakpoint = childWidthArray.length - 1
+  let breakpoint = childWidthArray.length // assume all items will fit
   let sumsOfChildWidth = 0
   for (const [index, childWidth] of childWidthArray.entries()) {
+    sumsOfChildWidth = sumsOfChildWidth + childWidth.width + GAP
     if (sumsOfChildWidth > widthToFit) {
-      breakpoint = index - 1
+      breakpoint = index
       break
     } else {
-      // The the gap between items into account when calculating the number of items possible
-      sumsOfChildWidth = sumsOfChildWidth + childWidth.width + GAP
+      continue
```

Comment:
I _think_ calculating the sum first and then to see if that will fit into the available width is a better approach. It fixes [this specific overflow issue](https://github.com/primer/react/issues/2865#issuecomment-1602426790)